code/modules/almayer/weaponhits.dm

```diff
@@ -15,109 +13,47 @@
-
-/proc/weaponhits_effects(weaponused, weaponmiss = FALSE, shotspacing = 0)
-	switch(weaponused)
-		if(WEAPON_MISSILE)
-			if(!weaponmiss)
-				for(var/mob/living/carbon/current_mob in GLOB.living_mob_list)
-					if(!is_mainship_level(current_mob.z))
-						continue
-					playsound_client(current_mob.client, 'sound/effects/metal_crash.ogg', 100 )
-					playsound_client(current_mob.client, 'sound/effects/bigboom3.ogg', 100)
-					addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(playsound_client), current_mob.client, 'sound/effects/pry2.ogg', 20), 1 SECONDS)
-					addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(playsound_client), current_mob.client, 'sound/effects/double_klaxon.ogg'), 2 SECONDS)
+/proc/weaponhits(weaponused, location, point_defense = FALSE)
+	var/hitchance = HIT_CHANCE_CHEAT
+	if(point_defense)
+		hitchance = HIT_CHANCE_STANDARD
+
+	var/datum/cause_data/cause_data = create_cause_data(weaponused ? "Railgun shot" : "Anti-Ship missile")
+	var/hit
+	var/list/echo_list = new(18)
+	if(weaponused)
+		for(var/turf/picked_atom in location)
+			if(prob(hitchance))
+				echo_list[ECHO_OBSTRUCTION] = -500
+				cell_explosion(picked_atom, 1000, 200, EXPLOSION_FALLOFF_SHAPE_EXPONENTIAL, null, cause_data)
+				shakeship(5, 5, FALSE, FALSE)
+				playsound(picked_atom, "bigboom", 50, 1, 200, echo = echo_list)
+				playsound(picked_atom, 'sound/effects/railgunhit.ogg', 50, 1, 200, echo = echo_list)
+				addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(playsound), picked_atom, 'sound/effects/double_klaxon.ogg', 25, 1, 200, VOLUME_SFX, 0, null, 1, echo_list), 2 SECONDS)
+				hit++
 			else
-				for(var/mob/living/carbon/current_mob in GLOB.living_mob_list)
-					if(!is_mainship_level(current_mob.z))
-						continue
-					addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(playsound_client), current_mob.client, 'sound/effects/laser_point_defence_success.ogg', 100), shotspacing SECONDS)
-					addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(to_chat), current_mob.client, SPAN_DANGER("You hear the Point Defense systems shooting down a missile!")), shotspacing SECONDS)
-
-		if(WEAPON_RAILGUN)
-			if(!weaponmiss)
-				for(var/mob/living/carbon/current_mob in GLOB.living_mob_list)
-					if(!is_mainship_level(current_mob.z))
-						continue
-					playsound_client(current_mob.client, 'sound/effects/bigboom3.ogg', 50)
-					playsound_client(current_mob.client, 'sound/effects/railgunhit.ogg', 50)
-					addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(playsound_client), current_mob.client, 'sound/effects/double_klaxon.ogg'), 2 SECONDS)
+				echo_list[ECHO_OBSTRUCTION] = -5000
+				addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(playsound), picked_atom, 'sound/effects/railgun_miss.ogg', 5, 1, 100, VOLUME_SFX, 0, null, 1, echo_list), 2 SECONDS)
+				addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(playsound), picked_atom, 'sound/effects/laser_point_defence_success.ogg', 15, 1, 100, VOLUME_SFX, 0, null, 1, echo_list), 2 SECONDS)
+			sleep(2 SECONDS)
```

Comment:
You do not need to sleep, just increase the callback timers for each volley and have the final announcement do the same.